src/util/psbt/error.rs

```diff
@@ -76,6 +76,9 @@ pub enum Error {
     MergeConflict(String),
     /// Serialization error in bitcoin consensus-encoded structures
     ConsensusEncoding,
+    #[cfg(feature = "base64")]
+    /// Base58 encoding error of PSBT data
+    Base64(base64::DecodeError),
```

Comment:
You cannot have a feature-gated error variant, in a major release or otherwise. This will always cause compilation breaks for downstream users that they cannot really control.

Reply:
Interesting, never thought about that, but probably because cargo uses the union of all features? So a new dependency that activates it can wreak havoc without exposing the relevant types, so it's legal for them to change the feature set without major version bumps.

Reply:
@apoelstra yes, and this is exactly the reason why this change is not a part of this PR! As we discussing above, this PR includes last commit https://github.com/rust-bitcoin/rust-bitcoin/pull/557/commits/b4077213ab311bed2a4faaaae3600d9949b867a6 with which this is not an API break. This commit has to be reversed after the minor version release such that with the next major version we will get this error type improvement (but not before that).

Reply:
Well, it appears we can't even do that. Let's imagine crate `A` depends on `rust-bitcoin` without features and crate `B` which also depends on rust-bitcoin without features, but doesn't expose it externally. Now `B` decides to turn on `base64` and does so in a minor release since it doesn't break the API (rust-bitcoin is a purely internal dependency). Now what cargo does is, when compiling one `rust-bitcoin` for both `A` and `B`, it calculates the union of all activated features in the dependency tree and uses that, breaking any previously exhaustive matches in `A`.

Reply:
This will be the case when feature gate get introduced. In the current PR it is not: it will be there only after the revert of the last commit

Reply:
Well, afaik @apoelstra likes to review PRs commit-by-commit. Having this in there now means it won't be a clean revert but rather another change too. So there's really no point in having this feature-gating in the history and risking leaving it in when reverting.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/pull/557#issuecomment-782033192